### ISSUE TITLE
CustomFuseGraph does not work on nested blocks

### ISSUE THREAD

**Issue by jjsjann123** [pytorch/pytorch#22734]
## 🐛 Bug

GraphFuser does not pass its FusionCallback and Symbol to recursive calls when fusing nested blocks: https://github.com/pytorch/pytorch/blob/d21e476dcd38fc61897812417f41ebfc6f2dd8cd/torch/csrc/jit/passes/graph_fuser.cpp#L1158

## To Reproduce

I was using the compiler backend tutorial repo here: https://github.com/bwasti/pytorch_compiler_tutorial

If you modify their `test.py` to put the graph in a nested scope:
modifying this
```
def foo(a, b):
  c = a.mul(b)
  a = c.mul(c)
  a = c.mul(a)
  return a
```
to
```
@torch.jit.script
def foo_compiled(a, b):
  if a.size(0) == b.size(0):
    c = a.mul(b)
    a = c.mul(c)
    a = c.mul(a)
  return a
```

The graph fuser can't fuse the aten::mul ops any more.
```
root@192c65b6e9e6:/volume/pytorch_compiler_tutorial# PYTHONPATH=build python test.py
-- Default IR --
 graph(%a.1 : Float(*),
      %b.1 : Float(*)):
  %2 : int = prim::Constant[value=0]() # test.py:20:13
  %3 : int = aten::size(%a.1, %2) # test.py:20:6
  %4 : int = aten::size(%b.1, %2) # test.py:20:19
  %5 : bool = aten::eq(%3, %4) # test.py:20:6
  %a : Float(*) = prim::If(%5) # test.py:20:3
    block0():
      %c.1 : Float(*) = aten::mul(%a.1, %b.1) # test.py:21:9
      %a.4 : Float(*) = aten::mul(%c.1, %c.1) # test.py:22:9
      %a.7 : Float(*) = aten::mul(%c.1, %a.4) # test.py:23:9
      -> (%a.7)
    block1():
      -> (%a.1)
  return (%a)

Default version took 1.12ms

-- Transformed IR --
 graph(%a.1 : Float(*),
      %b.1 : Float(*)):
  %2 : int = prim::Constant[value=0]() # test.py:35:13
  %3 : int = aten::size(%a.1, %2) # test.py:35:6
  %4 : int = aten::size(%b.1, %2) # test.py:35:19
  %5 : bool = aten::eq(%3, %4) # test.py:35:6
  %a : Float(*) = prim::If(%5) # test.py:35:3
    block0():
      %c.1 : Float(*) = aten::mul(%a.1, %b.1) # test.py:36:9
      %a.4 : Float(*) = aten::mul(%c.1, %c.1) # test.py:37:9
      %a.7 : Float(*) = aten::mul(%c.1, %a.4) # test.py:38:9
      -> (%a.7)
    block1():
      -> (%a.1)
  return (%a)

Compiled version took 1.16ms
```
## Expected behavior

We should be able to recognize this if we pass in FusionCallback & Symbol in our nested loop.

## Environment

Verified on ToT master. Commit can be seen on the link posted above.

**Comment by jjsjann123** [comment]
cc @ngimel @csarofeen

**Comment by suo** [comment]
Seems like this was fixed by #22775
